go.mod

```diff
@@ -1,9 +1,8 @@
 module sigs.k8s.io/kube-api-linter
 
-go 1.23.0
+go 1.24.0
```

Comment:
```suggestion
go 1.24.2
```

Reply:
Why do we need the patch version? I've never understood quite why a go mod would ever not just be on the 0 patch for each release since we are adding the 1.24 so that we can use 1.24 features, but 1.24.2 won't have introduced new language features so why do we need to require the newer patch version

I guess one motivation for keeping it updated with newer patch versions is that we are depending on this for the CI automation too?

Reply:
After releasing gotoolchain in Go1.21, we can specify the Go version more strictly that user uses. When we specify Go1.24.2, the developer run the program as Go1.24.2 even if the user uses lower Go version.
The larger patch version is good than lower patch as a security and performance etc. I don't have strong opinion to force to upgrade Go1.24.2, but I think it's better to use latest version if there is no reason since the dependency also is little.

Reply:
Ack, that makes sense. I'll merge this PR as is, but I'm not going to object if you include a bump from 1.24.0 to 1.24.2 in #63 then I won't complain ;)